src/coreclr/jit/codegenxarch.cpp

```diff
@@ -6591,8 +6590,9 @@ void CodeGen::genCkfinite(GenTree* treeNode)
     // Copy the floating-point value to an integer register. If we copied a float to a long, then
     // right-shift the value so the high 32 bits of the floating-point value sit in the low 32
     // bits of the integer register.
-    instruction ins = ins_CopyFloatToInt(targetType, (targetType == TYP_FLOAT) ? TYP_INT : TYP_LONG);
-    inst_RV_RV(ins, op1->GetRegNum(), tmpReg, targetType);
+    regNumber srcReg        = op1->GetRegNum();
+    var_types targetIntType = ((targetType == TYP_FLOAT) ? TYP_INT : TYP_LONG);
+    inst_RV_RV(ins_Copy(srcReg, targetIntType), tmpReg, srcReg, targetType);
```

Comment:
Not related to your PR, but it looks like we could optimize `Ckfinite` to just use the relevant `SSE`/`SSE2`/`AdvSimd` comparisons (it should just be `andps`, `ucomiss`, `setne`)

Reply:
could you please open an issue about it?

Reply:
Logged https://github.com/dotnet/runtime/issues/48022